presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -191,6 +191,16 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
      */
     BlockEncoding getEncoding();
 
+    /**
+     * Create a new block from the current block by keeping the same elements
+     * only with respect to {@code visiblePositions}.
+     * May return a view over the data in this block or may return a copy
+     */
+    default Block getPositions(int[] positions)
```

Comment:
Maybe use more meaningful name here, like `mask` to avoid confusion if this copies or masks posistions

Reply:
Oh, I see comment here: https://github.com/prestodb/presto/pull/8629#discussion_r152436350. nvm.